### PR TITLE
Prep initial charts indexing

### LINKED ISSUE #7: Update each chart's README to add new helm repo in installation instructions
Ideally this should happen before indexing/releasing first version of each chart, otherwise when aggregated in the hubs the README will refer `stable` instead of the new `prometheus-community` repo.

### LINKED ISSUE #8: Deprecate prometheus-operator chart before helm repo index, so that it won't be listed in the hubs
This chart will be renamed, but first must be deprecated before the prometheus-community/helm-charts repo is indexed, so that it won't be listed in the hubs. See [this prometheus-community issue](https://github.com/prometheus-community/community/issues/28#issuecomment-670406329) for reasoning and next steps.

### PR DESCRIPTION
**Pull request by scottrigby**
Fixes #7, Fixes #8

- [x] Cherry-pick commits from the one PR that was merged since the code was copied https://github.com/helm/charts/pull/23506 (see #13 for context)
- [x] Deprecate charts/prometheus-operator (see #8 and #9 for context)
- [x] Remove prow `OWNERS` files (see #12 for context)

Charts that need to be prepared for initial helm index
- [x] charts/prometheus
- [x] charts/prometheus-adapter
- [x] charts/prometheus-blackbox-exporter
- [x] charts/prometheus-cloudwatch-exporter
- [x] charts/prometheus-consul-exporter
- [x] charts/prometheus-couchdb-exporter
- [x] charts/prometheus-mongodb-exporter
- [x] charts/prometheus-mysql-exporter
- [x] charts/prometheus-nats-exporter
- [x] charts/prometheus-node-exporter
- [x] charts/prometheus-postgres-exporter
- [x] charts/prometheus-pushgateway
- [x] charts/prometheus-rabbitmq-exporter
- [x] charts/prometheus-redis-exporter
- [x] charts/prometheus-snmp-exporter
- [x] charts/prometheus-to-sd

**Suggestion:** per the steps in the plan at https://github.com/prometheus-community/community/issues/28#issuecomment-670230344 (which has been approved for the prometheus-community), for this PR review and comments let's discuss only what should happen in the intial step. To be clear, this PR is the first step, before "Step 2. Collaborate with prometheus devs to refactor/rename and simplify charts as appropriate." I outlined what I think we need in #7, but please let me know if I'm missing anything.

## 📊  Maintainers poll
How do you all feel about the new `charts/prometheus` README update in b324685? I Needed to update references to stable repo, but took the opportunity to reorganize, fix and simplify. Is this a good pattern for the other chart READMEs? Note major differences are:
- Added Helm 2 and Helm 3 commands, with links to Helm 3 command reference links (from there, users can select Helm 2 from the version dropdown if they wish)
- Removed the options table that's duplicated from (often out of sync with) `values.yaml` in favor of `helm inspect` (Helm 2) and `helm show values` (Helm 3)

**Results:** new template has been approved by all who responded

### CODE REVIEW

**Comment by Miouge1** [comment]
LGTM, I think it makes sense to drop the table of values from the README.md, it's often hard to keep it in sync with reality.

**Comment by scottrigby** [comment]
So far it sounds like we may be able to compromise with `helm upgrade [RELEASE_NAME] [CHART] --install`. Welcome other opinions here as well (though not trying to bike-shed). Looking for feedback on the readme reorganization overall as a pattern for the rest of the chart updates. Thanks!

**Comment by monotek** [comment]
I'm ok with using the upgrade command. Maybe the long version (--install instead of -i) of the arguments should be used to be clear.

Rest of the readme looks good to me and could be used as a template for the other charts.

**Comment by scottrigby** [comment]
@monotek @steven-sheehy & gang, I put the Helm 3 commands above Helm 2. And instead of using `helm upgrade` for installation instructions, I took suggestions above for making `helm install` use an explicit release name, then added `helm upgrade` commands in a similar way to the "Upgrading chart" section so there's no confusion. I also reorganized the upgrading MAJOR version headers slightly to be more clear to end users, depending on the version they're upgrading to. This has worked out well for other community charts. The pattern is like this:

```markdown
## Upgrading Chart
(helm commands)

## To v6.0
(instructions)

## To v5.0
(instructions)

## To v4.0
(instructions)

## From `foo` legacy chart located <HERE> (if any)
(instructions)
```

Does this look reasonable? I can apply this same pattern to the other charts.

**Review by monotek — APPROVED** [review]

File: charts/prometheus/README.md
Comment (Xtigyro) [inline charts/prometheus/README.md:49]: Why not just `helm delete` for both Helm versions? In Helm v3 `helm del` actually purges the chart by default, right?

File: charts/prometheus/README.md
Comment (Xtigyro) [inline charts/prometheus/README.md:14]: the `$` shouldn't be there if there isn't a real output after exec-ing the cmd.

Comment (Xtigyro) [reply charts/prometheus/README.md:14]: The same for all the `$` additions.

**Review by Xtigyro — CHANGES_REQUESTED** [review]

File: charts/prometheus/README.md
Comment (scottrigby) [reply charts/prometheus/README.md:49]: For Helm 2, `--purge` is recommended. In Helm 3 `delete` is an alias for `uninstall`, but has no `--purge` flag. Therefore it can't be identical.

File: charts/prometheus/README.md
Comment (scottrigby) [reply charts/prometheus/README.md:14]: I just added for consistency between all console blocks, so it's unambiguous to end users which parts of codeblocks are commands, and which refer to output or otherwise. Does that makes sense?

Comment (Xtigyro) [reply charts/prometheus/README.md:14]: Yeah - but if there's no output, there should be no `$` according to the MD linter.

File: charts/prometheus/README.md
Comment (scottrigby) [reply charts/prometheus/README.md:49]: Follow-up from chat: `delete` was added as an alias to ease some CI workflows, but it's not encouraged for Helm 3 generally, which is why it's not in the docs. The Helm `uninstall` concept itself is different now with Helm 3 architectural changes than default Helm 2 `delete` , but in the end maps to Helm 2's helm `delete --purge`.

Mainly we are trying to help lead users to Helm 3, since there is only now 3 months until Helm 2 is unsupported.

To be clear, `delete` can be used with Helm 3, but not encouraged due to user confusion between the different Helm versions, when `--purge` is needed or not for a full uninstall, leading to different results depending on version. It's a lot to explain fully so IMO easiest to document here as different commands for Helm 3 and 2, then just remove Helm 2 references after Helm 2 support ends.

You already said in Slack that makes sense, but just so others who may be interested can catch up as well 😄

File: charts/prometheus/README.md
Comment (scottrigby) [reply charts/prometheus/README.md:14]: That makes sense. My feeling is, the only real reason to have multiline shell commands with no $ prefix is for multi-line copy/paste. This is how I interpret the linting rule rationale you posted above as well. I agree in this case it's a good idea to copy/paste as a block - changing! Thanks for the detailed review.

**Comment by scottrigby** [comment]
I've now resolved all prometheus chart maintainer feedback, so will apply this pattern to the other charts so we can get this merged and expedite the repo transfer ⏳

**Review by Xtigyro — APPROVED** [review]

File: charts/prometheus/README.md
Comment (Xtigyro) [inline charts/prometheus/README.md:172]: Can we include please only working examples - just noticed this one?

**Review by Xtigyro — CHANGES_REQUESTED** [review]

File: charts/prometheus/README.md
Comment (scottrigby) [reply charts/prometheus/README.md]: Fixed in adb4e21

**Review by Xtigyro — APPROVED** [review]

**Comment by scottrigby** [comment]
Tests were failing for `prometheus-couchdb-exporter` (example [lint-test](https://github.com/scottrigby/prometheus-helm-charts/pull/14/checks?check_run_id=1005193296) GH Actions job). Had to fix k8s 1.16 deprecated PodSecurityPolicy in the extensions/v1beta1 API version. See https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ I'm guessing this chart hasn't been modified in a while 😛

**Comment by scottrigby** [comment]
**Update:** All initial chart readme changes have been pushed. Waiting on tests. Please give this a once over review. Once this is approved, we'll merge and move ahead 😅

File: charts/prometheus-couchdb-exporter/README.md
Comment (gkarthiks) [inline charts/prometheus-couchdb-exporter/README.md:15]: @scottrigby do you still want to have the stable here?

**Review by zanhsieh — APPROVED** [review]

**Review by okgolove — APPROVED** [review]

File: charts/prometheus-couchdb-exporter/README.md
Comment (scottrigby) [reply charts/prometheus-couchdb-exporter/README.md]: Good catch. No we only want this on the prometheus chart, since only it and the deprecated still older named `prometheus-operator` chart (see #9) have dependencies in the stable repo.

**Comment by scottrigby** [comment]
PS re-running gh action job. This appears to be an API blip:
>  ✖︎ prometheus-consul-exporter => (version: "0.1.6", path: "charts/prometheus-consul-exporter") > Error validating maintainer 'timm088': 502 Bad Gateway

Just tested locally and it validates fine:
```console
$ docker run --rm -it -v $(pwd):/app:cached quay.io/helmpack/chart-testing:v3.0.0
$ / #
$ / # cd /app
$ /app # ct lint --charts /app/charts/prometheus-consul-exporter/
...
1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ prometheus-consul-exporter => (version: "0.1.7", path: "/app/charts/prometheus-consul-exporter/")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

However, now getting similar errors:
>  ✖︎ prometheus => (version: "11.12.1", path: "charts/prometheus") > Error validating maintainer 'gianrubio': 429 too many requests

According to GitHub everything is fine https://www.githubstatus.com/ 🤔 

We may have to just try to run the job again 🤦 ⏳

**Comment by scottrigby** [comment]
😑 https://github.com/scottrigby/prometheus-helm-charts/runs/1007003791

I've honestly no idea which GH user `GITHUB_TOKEN` ties to for [GH API rate limits](https://developer.github.com/v3/rate_limit/). Don't seem to see this in [the docs for this token](https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#about-the-github_token-secret) in actions. It doesn't seem to be my user though:
```
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4992
X-RateLimit-Reset: 1597920256
```

Fine, I'll go back to sleep and re-run later 😴 

**PS:** If anyone wants to join us at the [Deep Dive Into Helm](https://kccnceu20.sched.com/event/Zx4K/deep-dive-into-helm-scott-rigby-codeacademy-paul-czarkowski-pivotal) session today, it starts in a little over 2 hours 🕣 See some of you then!

**Comment by scottrigby** [comment]
Woot! tests have passed! Let's get to the next steps. Not all maintainers of these charts have participated yet, but they may or may not be available. The goal of this PR is to change at least the charts readmes so the initial helm repo indexing will contain only updated info, and relevant to the new repo (we had to fix things along the way only because some of these charts have not been touched for some time). In future PRs, they will only be merged once at least one maintainer of the affected charts approve the PR. For that reason I recommend future PRs try to touch only one chart at a time like we do in the stable and incubator git source repo (github.com/helm/charts).

**Comment by scottrigby** [comment]
**Update:** Maintainers please check https://github.com/scottrigby/prometheus-helm-charts/issues/11 for another poll to see if there are any remaining blockers for the repo transfer to `prometheus-community` :shipit: 

There are also testing steps in that issue, if anyone wants to try it out now 🚀

**Review by gkarthiks — COMMENTED** [review]
approved

**Pull request merged**
